app/app.go

```diff
@@ -406,7 +406,7 @@ func NewRegenApp(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest
 	// NOTE: capability module's beginblocker must come before any modules using capabilities (e.g. IBC)
 	app.mm.SetOrderBeginBlockers(
 		upgradetypes.ModuleName, capabilitytypes.ModuleName, minttypes.ModuleName, distrtypes.ModuleName, slashingtypes.ModuleName,
-		evidencetypes.ModuleName, stakingtypes.ModuleName, ibchost.ModuleName,
+		evidencetypes.ModuleName, stakingtypes.ModuleName, ibchost.ModuleName, /* ecocredit.ModuleName, */
```

Comment:
Any ideas why adding this would cause the tests to hang?

Reply:
It times out after 10 minutes with the following error:
```
    tx.go:102: 
                Error Trace:    tx.go:102
                                                        suite.go:118
                                                        tx.go:50
                                                        modules_test.go:12
                Error:          Received unexpected error:
                                rpc error: code = NotFound desc = rpc error: code = NotFound desc = account regen1huayfhrzpkxxws60dxlzv6p3l6e95nau7yk3as not found: key not found
                Test:           TestEcocreditIntegration
panic: test timed out after 10m0s
```

Reply:
i'm wondering if this has anything to do with the ecocredit module not fulfilling the interface that actually contains the BeginBlock signature. 

might need to : `var _ module.AppModule = Module{}` and implement the methods it says its missing. maybe that would work?

Reply:
I think it might be best to break this out into a separate issue. The ecocredit module does not implement the `AppModule` interface and in attempting to do so there are some conflicts with the `RegisterServices` method which uses a different `Configurator`. Opened #689.